src/sort/sort.ts

```diff
@@ -0,0 +1,52 @@
+import { Style, Layer } from 'mapbox-gl'
+import { Dictionary, ExtendedStyle, ExtendedLayer, Group } from '../types/types'
+
+const GROUP_ORDER = [
+  Group.Background,
+  Group.Basemap,
+  Group.Heatmap,
+  Group.OutlinePolygons,
+  Group.Default,
+  Group.Point,
+  Group.Track,
+  Group.BasemapForeground,
+  Group.Label,
+  Group.Overlay,
+]
+
+const GROUP_FROM_LEGACY: Dictionary<Group> = {
+  'basemap-background': Group.Basemap,
+  static: Group.OutlinePolygons,
+  temporal: Group.Point,
+  tracks: Group.Track,
+  'basemap-foreground': Group.BasemapForeground,
+  tools: Group.Overlay,
+}
+
+export const convertLegacyGroups = (style: Style): ExtendedStyle => {
+  const newStyle = { ...style }
+  newStyle.layers =
+    newStyle.layers &&
+    newStyle.layers.map((layer: Layer) => {
+      const legacyGroup = layer.metadata && layer.metadata['mapbox:group']
+      if (legacyGroup) {
+        layer.metadata.group = GROUP_FROM_LEGACY[legacyGroup]
+      }
+      return layer
+    })
+  return newStyle
+}
+
+const sort = (style: ExtendedStyle): ExtendedStyle => {
+  const newStyle = { ...style }
+  newStyle.layers = newStyle.layers?.sort((a: ExtendedLayer, b: ExtendedLayer) => {
+    const aGroup = a.metadata?.group || Group.Default
+    const bGroup = b.metadata?.group || Group.Default
+    const aPos = GROUP_ORDER.indexOf(aGroup)
```

Comment:
This means the GROUP_ORDER is fixed, isn't it? what about use it as default selection but also export the list to be able to customise from outside ?

Reply:
It's a good idea but I'd prefer waiting for the use case to present itself before implementing.